vscode/microsoft-kiota/src/extension.ts

```diff
@@ -60,21 +57,53 @@ export async function activate(
   await loadTreeView(context);
   await checkForLockFileAndPrompt(context);
   let codeLensProvider = new CodeLensProvider();
+  let deepLinkParams: Record<string, string|undefined> = {};
   context.subscriptions.push(
     vscode.window.registerUriHandler({
       handleUri: async (uri: vscode.Uri) => {
         if (uri.path === "/") {
           return;
         }
+        console.log(uri);
         const queryParameters = getQueryParameters(uri);
+        console.log(queryParameters);
         if (uri.path.toLowerCase() === "/opendescription") {
           reporter.sendTelemetryEvent("DeepLink.OpenDescription");
           const descriptionUrl = queryParameters["descriptionurl"];
+          const name = getSanitizedString(queryParameters["name"]);
+          const source = getSanitizedString(queryParameters["source"]);
+          reporter.sendTelemetryEvent("DeepLink.OpenDescription", {"source": source}); 
+          let lowercasedKind: string = queryParameters["kind"]?.toLowerCase();
+          let validKind: string | undefined = ["plugin", "client"].indexOf(lowercasedKind) > -1 ? lowercasedKind : undefined ;
+          if (!validKind){
+            vscode.window.showInformationMessage("Invalid parameter 'kind' deeplinked.");
+          }
+          let acceptedLanguages: string [] = Object.keys(KiotaGenerationLanguage).filter(x => !Number(x)&& x !=='0');
+          let givenLanguage:string | undefined = acceptedLanguages.indexOf(queryParameters["language"]) > -1 ? queryParameters["language"] : undefined ;
+          if (!givenLanguage && validKind === "client"){
```

Comment:
I think we could leverage TypeScript's [enum reverse mappings](https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings) to simplify the code here. That way, we could write:

```suggestion
          const givenLanguage = KiotaGenerationLanguage[queryParameters["language"]];
          if (!givenLanguage && validKind === "client"){
```

Reply:
Goal is not to get the enum parsed or the numbered enum value but to retain the string as is but verify the string is among accepted values.

Reply:
> Are the accepted languages different from the enum entries

Yes. I want the key values themselves. Because they will be used elsewhere as is. Using enum values to parse will give me the enumerators 0,1,2 etc as returned values which I can't carry forward. I need the language value itself going forward. E.g "Go" not "5"

Reply:
You can get the string back if you want. Just use `MyEnum[0]` to get the string `"A"`